src/internal/connector/exchange/service_restore.go

```diff
@@ -313,7 +315,7 @@ func RestoreExchangeDataCollections(
 
 		containerID, err := GetContainerIDFromCache(
 			ctx,
-			gs,
+			creds,
```

Comment:
This is a midway point to transitioning away from passing around a service and towards passing credentials that can produce a service (and in the long run, passing a interface compliant struct with these details).  Thus the variations of passing both creds and service structs.

I do need to review service behavior gotchas with you, because I designed this on the assumption that lazily generating services is safe to do, and I figured that I could take that part back to the drawing board if necessary.  Sounds like my assumption is off the mark a little bit.